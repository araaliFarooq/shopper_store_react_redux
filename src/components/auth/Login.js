import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import cart from "../../assets/images/shoperwhite.png";
import { login } from "../../actions/LoginAction";
import { ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      errors: {}
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();
    const userData = {
      user_name: this.state.username,
      password: this.state.password
    };
    this.props.login(userData);
  };

  render() {
    return (
      <section id="contain">
        <div className="login_box">
          <div className="logos">
            <img src={cart} />
          </div>
          <div className="credits">
            <div className="heading" id="my-alert">
              <p>
                Access our store here just by loging in.
                <br />
                Fill in your credentials here.
              </p>
            </div>
            <form className="logs" id="loginform" onSubmit={this.onSubmit}>
              <div className="input-box">
                <input
                  type="text"
                  name="username"
                  id="username"
                  placeholder="username"
                  required=""
                  onChange={this.onChange}
                />
              </div>
              <div className="input-box">
                <input
                  type="password"
                  name="password"
                  id="password"
                  placeholder="password"
                  required=""
                  onChange={this.onChange}
                />
              </div>
              <div className="input-box">
                <input type="submit" name="" value="Log In" />
              </div>
              <p />
              <p>
                <small>
                  <a href="all_products.html">
                    Use this link to go to the home page
                  </a>
                </small>
              </p>
            </form>
          </div>
        </div>
        <ToastContainer autoClose={13000} />
      </section>
    );
  }
}
Login.propTypes = {
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
  errors: PropTypes.object,
  login: PropTypes.func
};

const mapStateToProps = state => ({
  errors: state.loginReducer.errors
});

export default connect(
  mapStateToProps,
  { login }
)(Login);
