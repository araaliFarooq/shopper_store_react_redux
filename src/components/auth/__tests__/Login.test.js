import React from "react";
import expect from "expect";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import Login from "../Login";

const storeFake = (state, action) => ({
  subscribe: jest.fn(),
  default: jest.fn(),
  getState: () => state,
  dispatch: () => action
});

const store = storeFake({
  loginReducer: {
    errors: {},
    user: {}
  }
});

describe("Login functionality", () => {
  const props = {};
  const wrapper = mount(
    <Provider store={store}>
      <Login {...props} />
    </Provider>
  );
  it("login should render correctly", () => {
    const component = wrapper;
    expect(component).toMatchSnapshot();
  });

  it("listens for submit", () => {
    const url = window.location;
    const arr = url.toString().split("/");
    const form = wrapper.find("#loginform");
    form.simulate("submit");
    expect(window.location.href).toEqual(`${arr[0]}//${arr[2]}/`);
  });

  it("listens for state changes on the add product Form", () => {
    const event = { target: { name: "testName", value: "testValue" } };
    const product = wrapper.find("#username").first();
    product.value = "araali";
    product.simulate("change", event);
    expect(product.value).toEqual("araali");
  });
});
