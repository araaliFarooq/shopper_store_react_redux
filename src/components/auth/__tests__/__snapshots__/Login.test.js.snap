// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login functionality login should render correctly 1`] = `ReactWrapper {}`;
